helpick-service/src/main/java/com/infoshare/service/NeedRequestService.java

```diff
@@ -26,27 +35,82 @@ public NeedRequestService(DB db) {
+            for (int i = 0; i < activeNeedRequests.size(); i++) {
+                if (activeNeedRequests.get(i).getUuid().equals(changedRequest.getUuid())) {
+                    activeNeedRequests.set(i, changedRequest);
+                }
+            }
+            FileWriter writer = new FileWriter("NeedRequest.csv", false);
+            for (NeedRequest nr : activeNeedRequests) {
+                db.saveNeedRequest(nr);
+            }
+            writer.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void updateRequestsStatus() {
+        try {
+            List<NeedRequest> activeNeedRequests = db.getNeedRequests();
+            for (NeedRequest request : activeNeedRequests) {
+                Date time1 = request.getStatusChange();
+                Date actualTime = new Date();
+                long diff = actualTime.getTime() - time1.getTime();
+                long minutes = TimeUnit.MILLISECONDS.toMinutes(diff);
+                if (request.getHelpStatus().equals(HelpStatuses.INPROGRESS) &&
+                        minutes > 1440) {
+                    log.info("Found difference > 24h in request " + request.getUuid() + ", changing status...");
+                    request.setHelpStatus(HelpStatuses.NEW);
+                    request.setStatusChange(new Date());
+                    log.info("Status of request ID " + request.getUuid() + "restored to NEW");
+                }
+            }
+            FileWriter writer = new FileWriter("NeedRequest.csv", false);
+            for (NeedRequest nr : activeNeedRequests) {
+                db.saveNeedRequest(nr);
+            }
+            writer.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void printNeedRequestsList(List<NeedRequest> needRequestList) {
+        for (int i = 0; i < needRequestList.size(); i++) {
+            System.out.println(i + ". " + needRequestList.get(i));
+        }
+    }
+
     public List<NeedRequest> getNeedRequestFilteredList(String city, TypeOfHelp typeOfHelp) {
         return db.getNeedRequests().stream()
-            .filter(n -> n.getHelpStatus().equals(HelpStatuses.NEW))
-            .filter(n -> n.getTypeOfHelp().equals(typeOfHelp))
-            .filter(n -> n.getPersonInNeed().getLocation().equalsIgnoreCase(city))
-            .collect(Collectors.toList());
+                .filter(n -> n.getHelpStatus().equals(HelpStatuses.NEW))
+                .filter(n -> n.getTypeOfHelp().equals(typeOfHelp))
+                .filter(n -> n.getPersonInNeed().getLocation().equalsIgnoreCase(city))
+                .collect(Collectors.toList());
     }
-    public Optional<NeedRequest>getNeedRequestById(UUID uuid){
-      return db.getNeedRequests().stream()
-      .filter(n->n.getUuid().equals(uuid))
-      .findAny();
+
+    public Optional<NeedRequest> getNeedRequestById(UUID uuid) {
```

Comment:
tutaj mozna zmienic na: 
```suggestion
    public NeedRequest getNeedRequestById(UUID uuid) {
        return db.getNeedRequests().stream()
                .filter(n -> n.getUuid().equals(uuid))
                .collect(Collectors.toList()).get(0);
    }
```

Reply:
Dzięki, ale zostawiam swoją wersję.